main/src/addins/MonoDevelop.DesignerSupport/MonoDevelop.DesignerSupport.Toolbox/MacToolboxViewItems.cs

```diff
@@ -0,0 +1,301 @@
+				context.SaveGraphicsState ();
+				ExpanderImage.Draw (new CGRect (Frame.Width - ExpanderImage.Size.Height - 5, 4, ExpanderImage.Size.Width, ExpanderImage.Size.Height));
+				context.RestoreGraphicsState ();
+			}
+		}
+
+		#region INativeChildView
+
+
+		public void OnKeyPressed (object o, Gtk.KeyPressEventArgs ev)
+		{
+
+		}
+
+		public void OnKeyReleased (object o, Gtk.KeyReleaseEventArgs ev)
+		{
+
+		}
+
+		#endregion
+	}
+
+	[Register ("ImageCollectionViewItem")]
+	class ImageCollectionViewItem : NSCollectionViewItem
+	{
+		internal static CGSize Size = new CGSize (23, 23);
+
+		internal const string Name = "ImageViewItem";
+		const int margin = 5;
+
+		public override bool Selected {
+			get => base.Selected;
+			set {
+				base.Selected = value;
+				if (contentCollectionView != null) {
+					contentCollectionView.IsSelected = value;
+				}
+				Refresh ();
+			}
+		}
+
+		public NSImage SelectedImage { get; set; }
+		public NSImage Image { get; set; }
+
+	 	public void Refresh ()
+		{
+			if (Selected) {
+				contentCollectionView.BackgroundImage = SelectedImage ;
+			} else {
+				contentCollectionView.BackgroundImage = Image;
+			}
+		}
+
+		public string AccessibilityTitle {
+			get => contentCollectionView.AccessibilityTitle;
+			set {
+				contentCollectionView.AccessibilityTitle = value;
+			}
+		}
+
+		public bool AccessibilityElement {
+			get => contentCollectionView.AccessibilityElement;
+			set {
+				contentCollectionView.AccessibilityElement = value;
+			}
+		}
+
+		ContentCollectionViewItem contentCollectionView;
+		public override void LoadView ()
+		{
+			View = contentCollectionView = new ContentCollectionViewItem ();
+			contentCollectionView.Identifier = Name;
+			contentCollectionView.EdgeInsets = new NSEdgeInsets (0, 0, 0, 0);
+		}
+
+		internal void SetCollectionView (NSCollectionView collectionView) => contentCollectionView.SetCollectionView (collectionView);
+
+		public ImageCollectionViewItem (IntPtr handle) : base (handle)
+		{
+
+		}
+	}
+
+	class ContentCollectionViewItem : NSStackView
+	{
+		public NSColor BackgroundSelectedColor { get; set; } = Styles.CellBackgroundSelectedColor;
+		public NSImage BackgroundImage { get; internal set; }
+
+		bool isSelected;
+		public bool IsSelected {
+			get => isSelected;
+			set {
+				if (isSelected == value) {
+					return;
+				}
+				isSelected = value;
+				NeedsDisplay = true;
+			}
+		}
+
+		MacToolboxWidget collectionView;
+		public ContentCollectionViewItem ()
+		{
+			Orientation = NSUserInterfaceLayoutOrientation.Horizontal;
+			TranslatesAutoresizingMaskIntoConstraints = false;
+
+			NeedsDisplay = true;
+		}
+
+		public override void ScrollWheel (NSEvent theEvent)
+		{
+			base.ScrollWheel (theEvent);
+			NeedsDisplay = true;
+		}
+
+		public override void DrawRect (CGRect dirtyRect)
+		{
+			base.DrawRect (dirtyRect);
+
+			if (isSelected) {
+				if (collectionView.IsFocused || collectionView == null) {
```

Comment:
```suggestion
				if (collectionView == null || collectionView.IsFocused) {
```

Otherwise this will NRE :)